dot/parachain/candidate-validation/messages.go

```diff
@@ -0,0 +1,196 @@
+	vdt := scale.VaryingDataType(*i)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting value to varying data type: %w", err)
+	}
+
+	*i = InvalidCandidate(vdt)
+	return nil
+}
+
+// Value returns the value from the underlying varying data type
+func (i *InvalidCandidate) Value() (scale.VaryingDataTypeValue, error) {
+	vdt := scale.VaryingDataType(*i)
+	return vdt.Value()
+}
+
+// ExecutionError failed to execute `valid_black`. This includes function panicking.
+type ExecutionError struct {
+	Error string
+}
+
+// Index returns the index of varying data type
+func (ExecutionError) Index() uint {
+	return 0
+}
+
+// InvalidOutputs validation outputs check doesn't pass
+type InvalidOutputs struct{}
+
+// Index returns the index of varying data type
+func (InvalidOutputs) Index() uint {
+	return 1
+}
+
+// Timeout Execution timeout
+type Timeout struct{}
+
+// Index returns the index of varying data type
+func (Timeout) Index() uint {
+	return 2
+}
+
+// ParamsTooLarge Validation input is over the limit
+type ParamsTooLarge struct {
+	ParamsSize uint64
+}
+
+// Index returns the index of varying data type
+func (ParamsTooLarge) Index() uint {
+	return 3
+}
+
+// CodeTooLarge code size is over the limit
+type CodeTooLarge struct {
+	CodeSize uint64
+}
+
+// Index returns the index of varying data type
+func (CodeTooLarge) Index() uint {
+	return 4
+}
+
+// PoVDecompressionFailure PoV does not decompress correctly
+type PoVDecompressionFailure struct{}
+
+// Index returns the index of varying data type
+func (PoVDecompressionFailure) Index() uint {
+	return 5
+}
+
+// BadReturn Validation function returned invalid data
+type BadReturn struct{}
+
+// Index returns the index of varying data type
+func (BadReturn) Index() uint {
+	return 6
+}
+
+// BadParent invalid relay chain parent
+type BadParent struct{}
+
+// Index returns the index of varying data type
+func (BadParent) Index() uint {
+	return 7
+}
+
+// PoVHashMismatch PoV hash mismatch
+type PoVHashMismatch struct{}
+
+// Index returns the index of varying data type
+func (PoVHashMismatch) Index() uint {
+	return 8
+}
+
+// BadSignature bod collator signature
+type BadSignature struct{}
+
+// Index returns the index of varying data type
+func (BadSignature) Index() uint {
+	return 9
+}
+
+// ParaHeadHashMismatch para head hash does not match
+type ParaHeadHashMismatch struct{}
+
+// Index returns the index of varying data type
+func (ParaHeadHashMismatch) Index() uint {
+	return 10
+}
+
+// CodeHashMismatch validation code hash does not match
+type CodeHashMismatch struct{}
+
+// Index returns the index of varying data type
+func (CodeHashMismatch) Index() uint {
+	return 11
+}
+
+// CommitmentHashMismatch validation has generated different candidate commitments
+type CommitmentHashMismatch struct{}
+
+// Index returns the index of varying data type
+func (CommitmentHashMismatch) Index() uint {
+	return 12
+}
```

Comment:
We don't need to Encode/Decode `InvalidCandidate`. 
Also, as you have removed your `ValidationResult` implementation and are using the existing one, the `InvalidCandidate` type will not be used anywhere.

As you can see in the existing `ValidationResult` struct, we have an error field in which we can also store the reason for invalidity.

Remove this type